vslib/src/CorePortIndexMap.cpp

```diff
@@ -0,0 +1,181 @@
+#include "CorePortIndexMap.h"
+
+#include "swss/logger.h"
+
+#include <set>
+
+using namespace saivs;
+
+CorePortIndexMap::CorePortIndexMap(
+        _In_ uint32_t switchIndex):
+    m_switchIndex(switchIndex)
+{
+    SWSS_LOG_ENTER();
+
+    // empty
+}
+
+uint32_t CorePortIndexMap::getSwitchIndex() const
+{
+    SWSS_LOG_ENTER();
+
+    return m_switchIndex;
+}
+
+bool CorePortIndexMap::add(
+        _In_ const std::string& ifname,
+        _In_ const std::vector<uint32_t>& corePortIndex)
+{
+    SWSS_LOG_ENTER();
+
+    auto n = corePortIndex.size();
+
+    if (n != 2)
+    {
+        SWSS_LOG_ERROR("Invalid corePortIndex. Core port index must have core and core port index %d, %s", n, ifname.c_str());
+        return false;
+    }
```

Comment:
For it is going to be only 2 fields. But not sure if this will change in the future. We may have switch_id (not switch index) as part of the mapping later.